solanaj/src/main/java/org/p2p/solanaj/kits/transaction/mapper/ConfirmedTransactionRootMapper.kt

```diff
@@ -0,0 +1,201 @@
+package org.p2p.solanaj.kits.transaction.mapper
+
+import org.p2p.solanaj.kits.transaction.BurnOrMintDetails
+import org.p2p.solanaj.kits.transaction.CloseAccountDetails
+import org.p2p.solanaj.kits.transaction.CreateAccountDetails
+import org.p2p.solanaj.kits.transaction.TransactionDetails
+import org.p2p.solanaj.kits.transaction.TransferDetails
+import org.p2p.solanaj.kits.transaction.UnknownDetails
+import org.p2p.solanaj.kits.transaction.network.ConfirmedTransactionRootResponse
+import org.p2p.solanaj.kits.transaction.network.meta.InstructionInfoDetailsResponse
+import org.p2p.solanaj.kits.transaction.network.meta.InstructionResponse
+import org.p2p.solanaj.kits.transaction.parser.OrcaSwapInstructionParser
+import org.p2p.solanaj.kits.transaction.parser.SerumSwapInstructionParser
+import org.p2p.solanaj.programs.SystemProgram.SPL_TOKEN_PROGRAM_ID
+
+// Get some parsing info from here
+// https://github.com/p2p-org/solana-swift/blob/main/Sources/SolanaSwift/Helpers/TransactionParser.swift#L74-L86
+internal class ConfirmedTransactionRootMapper(
+    private val orcaSwapInstructionParser: OrcaSwapInstructionParser,
+    private val serumSwapInstructionParser: SerumSwapInstructionParser
+) {
+
+    fun mapToDomain(
+        transactionRoot: ConfirmedTransactionRootResponse,
+        onErrorLogger: (Throwable) -> Unit
+    ): List<TransactionDetails> {
+        val signature = transactionRoot.transaction?.getTransactionId() ?: return emptyList()
+
+        return when {
+            orcaSwapInstructionParser.isTransactionContainsOrcaSwap(transactionRoot) -> {
```

Comment:
what if we create enumType of transactions like ORCA_SWAP_SERUM_SWAP to avoid create additional methods if we new types will be added

Reply:
```
when(orcaSwapInstructionParser.getTransactionType(transactionRoot)) {
    ORCA_SWAP -> //TODO
    SERUM_SWAP -> //TODO
    else -> //TODO
}
```

Reply:
if new swap type appears - we need to create new parsing logic anyways, so I think when without enum is ok here